Mathlib/SetTheory/Cardinal/Ordinal.lean

```diff
@@ -849,6 +849,54 @@ protected theorem eq_of_add_eq_add_right {a b c : Cardinal} (h : a + b = c + b)
   exact Cardinal.eq_of_add_eq_add_left h hb
 #align cardinal.eq_of_add_eq_add_right Cardinal.eq_of_add_eq_add_right
 
+section ciSup
+
+variable {ι : Type u} [Nonempty ι] {ι' : Type w} [Nonempty ι']
+variable (f : ι → Cardinal.{v}) (hf : BddAbove (range f)) (c : Cardinal.{v})
+
+protected theorem ciSup_add : (⨆ i, f i) + c = ⨆ i, f i + c := by
+  have : ∀ i, f i + c ≤ (⨆ i, f i) + c := fun i ↦ add_le_add_right (le_ciSup hf i) c
+  refine le_antisymm ?_ (ciSup_le' this)
+  have bdd : BddAbove (range (f · + c)) := ⟨_, forall_range_iff.mpr this⟩
+  obtain hs | hs := lt_or_le (⨆ i, f i) ℵ₀
+  · obtain ⟨i, hi⟩ := exists_eq_of_iSup_eq_of_not_isLimit
+      f hf _ (fun h ↦ hs.not_le h.aleph0_le) rfl
+    exact hi ▸ le_ciSup bdd i
+  rw [add_eq_max hs, max_le_iff]
+  exact ⟨ciSup_mono bdd fun i ↦ self_le_add_right _ c,
+    (self_le_add_left _ _).trans (le_ciSup bdd <| Classical.arbitrary ι)⟩
+
+protected theorem add_ciSup : c + (⨆ i, f i) = ⨆ i, c + f i := by
+  rw [add_comm, Cardinal.ciSup_add f hf]; simp_rw [add_comm]
+
+protected theorem ciSup_add_ciSup (g : ι' → Cardinal.{v}) (hg : BddAbove (range g)) :
+    (⨆ i, f i) + (⨆ j, g j) = ⨆ (i) (j), f i + g j := by
+  simp_rw [Cardinal.ciSup_add f hf, Cardinal.add_ciSup g hg]
+
+protected theorem ciSup_mul : (⨆ i, f i) * c = ⨆ i, f i * c := by
+  have : ∀ i, f i * c ≤ (⨆ i, f i) * c := fun i ↦ mul_le_mul_right' (le_ciSup hf i) c
```

Comment:
```suggestion
protected theorem ciSup_mul : (⨆ i, f i) * c = ⨆ i, f i * c := by
  by_cases hf : BddAbove (range f); swap
  · by_cases hc : c = 0; · simp [hc]
    show (dite _ _ _) * c = dite _ _ _
    rw [dif_neg, dif_neg, bot_eq_zero', zero_mul]
    · exact hf
    · exact fun hx ↦ hf ⟨⨆ i, f i * c, fun _ ⟨i, hi⟩ ↦ (le_mul_right hc).trans (hi ▸ le_ciSup hx i)⟩
  have : ∀ i, f i * c ≤ (⨆ i, f i) * c := fun i ↦ mul_le_mul_right' (le_ciSup hf i) c
```
This removes the `BddAbove` hypothesis.

Reply:
Thanks! I mostly copied the proof from `ciSup_add` but didn't check whether the hypotheses remain necessary. I think `Nonempty` can also be removed.